js/util.js

```diff
@@ -162,60 +165,55 @@ module.exports.asyncEventListener = function(listenerFun, logger) {
   }
 }
 
-/**
-   Extracts what listening address is closest to loopback. Returns 127.0.0.1 or equivalent
-   Ideally, or first hostname/IP in listening list otherwise.
-
-   Will throw if given an invalid config.
-*/
-module.exports.getLoopbackFromConfig = function(webappOptions) {
-  //if this isn't here, throw
-  let nodeConfig = webappOptions.serverConfig.node;
-
-  let getPreferredAddress = function(addresses) {
-    if (!addresses) {
-      return '127.0.0.1';
+module.exports.uniqueIps = Promise.coroutine(function *uniqueIps(hostnames) {
+  if (hostnames == null) {
+    loggers.network.debug("uniqueIps: no addresses specified, returning 0.0.0.0");
+    return [ '0.0.0.0' ];
+  }
+  let set = new Set();
+  for (let hostname of hostnames) {
+    try {
+      const ipAddress = yield dnsLookup(hostname);
+      set.add(ipAddress);
+    } catch (e) {
+      loggers.network.warn(`Skipping invalid listener address=${hostname}`);
     }
-    for (let i = 0; i < addresses.length; i++) {
-      let address = addresses[i];
-      if (address == 'localhost' || address == '0.0.0.0') {
```

Comment:
Note that we kinda did resolve one hostname here (localhost) but generally didn't resolve other hostnames